bundles/org.openhab.binding.modbus.kermi/src/main/java/org/openhab/binding/modbus/kermi/internal/handler/KermiXcenterThingHandler.java

```diff
@@ -0,0 +1,672 @@
+     */
+    protected volatile @Nullable ModbusCommunicationInterface comms = null;
+    private int slaveId;
+
+    public KermiXcenterThingHandler(Bridge thing) {
+        super(thing);
+
+        // STATE
+        globalStateIdChannel = channelUID(thing, STATE_GROUP, GLOBAL_STATE_ID_CHANNEL);
+
+        // ALARM
+        alarmStateChannel = channelUID(thing, ALARM_GROUP, ALARM_STATE_CHANNEL);
+
+        // Energy source
+        exitTemperatureChannel = channelUID(thing, ENERGYSOURCE_GROUP, EXIT_TEMPERATURE_CHANNEL);
+        incomingTemperatureChannel = channelUID(thing, ENERGYSOURCE_GROUP, INCOMING_TEMPERATURE_CHANNEL);
+        outsideTemperatureChannel = channelUID(thing, ENERGYSOURCE_GROUP, TEMPERATURE_SENSOR_OUTSIDE_CHANNEL);
+
+        // Loading circuit
+        flowTemperatureChannel = channelUID(thing, CHARGINGCIRCUIT_GROUP, FLOW_TEMPERATURE_CHANNEL);
+        returnFlowTemperatureChannel = channelUID(thing, CHARGINGCIRCUIT_GROUP, RETURN_TEMPERATURE_CHANNEL);
+        flowSpeedChannel = channelUID(thing, CHARGINGCIRCUIT_GROUP, FLOW_SPEED_CHANNEL);
+
+        // Power
+        copChannel = channelUID(thing, POWER_GROUP, COP_CHANNEL);
+        copHeatingChannel = channelUID(thing, POWER_GROUP, COP_HEATING_CHANNEL);
+        copDrinkingWaterChannel = channelUID(thing, POWER_GROUP, COP_DRINKINGWATER_CHANNEL);
+        copCoolingChannel = channelUID(thing, POWER_GROUP, COP_COOLING_CHANNEL);
+
+        powerChannel = channelUID(thing, POWER_GROUP, POWER_CHANNEL);
+        powerHeatingChannel = channelUID(thing, POWER_GROUP, POWER_HEATING_CHANNEL);
+        powerDrinkingWaterChannel = channelUID(thing, POWER_GROUP, POWER_DRINKINGWATER_CHANNEL);
+        powerCoolingChannel = channelUID(thing, POWER_GROUP, POWER_COOLING_CHANNEL);
+
+        electricPowerChannel = channelUID(thing, POWER_GROUP, ELECTRIC_POWER_CHANNEL);
+        electricPowerHeatingChannel = channelUID(thing, POWER_GROUP, ELECTRIC_POWER_HEATING_CHANNEL);
+        electricPowerDrinkingWaterChannel = channelUID(thing, POWER_GROUP, ELECTRIC_POWER_DRINKINGWATER_CHANNEL);
+        electricPowerCoolingChannel = channelUID(thing, POWER_GROUP, ELECTRIC_POWER_COOLING_CHANNEL);
+
+        // Work hours
+        workHoursFanChannel = channelUID(thing, WORKHOURS_GROUP, WORKHOURS_FAN_CHANNEL);
+        workHoursStorageLoadingPumpChannel = channelUID(thing, WORKHOURS_GROUP, WORKHOURS_STORAGE_LOADING_PUMP_CHANNEL);
+        workHoursCompressorChannel = channelUID(thing, WORKHOURS_GROUP, WORKHOURS_COMPRESSOR_CHANNEL);
+
+        // PV Modulation
+        pvStateChannel = channelUID(thing, PV_GROUP, PV_STATE_CHANNEL);
+        pvPowerChannel = channelUID(thing, PV_GROUP, PV_POWER_CHANNEL);
+        pvTargetTemperatureHeatingChannel = channelUID(thing, PV_GROUP, PV_TARGET_TEMPERATURE_HEATING_CHANNEL); // READ-WRITE
+        pvTargetTemperatureDrinkingWaterChannel = channelUID(thing, PV_GROUP,
+                PV_TARGET_TEMPERATURE_DRINKINGWATER_CHANNEL); // READ-WRITE
```

Comment:
This can be extract to `KermiBindingConstatns` typically bindings create  lists/set of channels/groups outside the handler. Maybe check some other bindings that use groups if you need an example.

Reply:
i followed the modbus-binding 'e3dc' from Bernd Weymann. But i'll review it as an improvement.